### PR TITLE
docs: update prow docs

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**Reason for Change**:

Refresh prow docs.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: .prowci/plugins.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 approve:
 - repos:
   - Azure/aks-engine
-  implicit_self_approve: true
   lgtm_acts_as_approve: true
+  require_self_approval: false
```
Comment (mboersma) [inline .prowci/plugins.yaml]: What behavior does this actually change? No "approve" label required?

Comment (CecileRobertMichon) [reply .prowci/plugins.yaml]: `implicit_self_approve: true` is deprecated and replaced by `require_self_approve: false`. No behavior change compared to before. The author does not need to approve their own PR.

https://github.com/kubernetes/test-infra/blob/master/prow/plugins/config.go#L233

**Pull request merged**